pkg/kubelet/cm/memorymanager/policy_static_test.go

```diff
@@ -4216,3 +4446,758 @@ func Test_isAffinityViolatingNUMAAllocations(t *testing.T) {
 		})
 	}
 }
+
+type containerOptions struct {
+	name          string
+	request       string
+	limit         string
+	restartPolicy *v1.ContainerRestartPolicy
+}
+
+type podOption func(*v1.Pod)
+
+func withPodResources(memRequest, memLimit string) podOption {
+	return func(pod *v1.Pod) {
+		pod.Spec.Resources = &v1.ResourceRequirements{
+			Requests: v1.ResourceList{
+				v1.ResourceMemory: resource.MustParse(memRequest),
+			},
+			Limits: v1.ResourceList{
+				v1.ResourceMemory: resource.MustParse(memLimit),
+			},
+		}
+	}
+}
+
+func makeMultiContainerPodWithOptions(initContainers, appContainers []*containerOptions, podOptions ...podOption) *v1.Pod {
```

Comment:
this is nice. In a future **unrelated** change we should unify all the helpers and remove redundancies